.github/workflows/test.yml

```
@@ -83,6 +106,23 @@ jobs:
       run: |
         pytest --cov --cov-report=term --cov-report=xml -v
 
+    - name: Assert that bundled version isn't built
+      if: ${{ matrix.os == 'ubuntu-latest' && matrix.tree-sitter-bash == 'system' }}
+      run: |
+        if [[ -n "$(find -name 'lang.so')" ]]; then
+          find -name 'lang.so'
+          echo "::error Bundled tree-sitter-bash was built"
+          false
+        fi
+
+    - name: Assert that bundled version is built
+      if: ${{ matrix.os == 'ubuntu-latest' && matrix.tree-sitter-bash == 'bundled' }}
+      run: |
+        if [[ ! -n "$(find -name 'lang.so')" ]]; then
+          echo "::error Bundled tree-sitter-bash was not built!"
+          false
+        fi
+
```

How about merging these two into one, something like (untested):

```suggestion
    - name: Check whether bundled version was built
      if: ${{ matrix.os == 'ubuntu-latest' && matrix.tree-sitter-bash == 'system' }}
      run: |
        EXPECTED=${{ matrix.tree-sitter-bash == 'bundled' }}  # assuming it gets converted to 0/1
        if [[ "$(find -name 'lang.so' | wc -l)" == ${EXPECTED} ]]; then
          find -name 'lang.so'
          echo "::error Expected ${{ matrix.tree-sitter-bash }} tree-sitter, found the other"
          false
        fi

```

That said, I'm not sure if expressions can be used in commands, you may need to push them via envvar instead. Also bools could be true/false rather than 1/0.

---

Done.